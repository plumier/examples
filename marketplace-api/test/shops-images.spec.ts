import { join } from "path"
import supertest from "supertest"

import createApp from "../src/app"
import { createShop, ignore } from "./_helper"


describe("Shops Images", () => {
    it("Should upload file properly", async () => {
        const app = await createApp({ mode: "production" })
        const { staffs, shop } = await createShop(app)

        const { body: img } = await supertest(app.callback())
            .post(`/api/v1/shops/${shop.id}/images/upload`)
            .attach("file", join(__dirname, "image/landscape.jpeg"))
            .set("Authorization", `Bearer ${staffs[0]?.token}`)
            .expect(200)
        const { body } = await supertest(app.callback())
            .get(`/api/v1/shops/${shop.id}/images/${img.id}`)
            .set("Authorization", `Bearer ${staffs[0]?.token}`)
            .expect(200)
        expect(body).toMatchSnapshot({ ...ignore, size: expect.any(Number) })
    })
})