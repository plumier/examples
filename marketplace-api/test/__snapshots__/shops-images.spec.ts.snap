// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shops Images Should upload file properly 1`] = `
Object {
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "mimeType": "image/jpeg",
  "name": "landscape.jpeg",
  "size": Any<Number>,
  "updatedAt": Any<String>,
  "url": "https://image.com/image.jpg",
}
`;
